builtInFunctions/esdtNFTCreate_test.go

```diff
@@ -437,10 +379,348 @@ func TestEsdtNFTCreate_ProcessBuiltinFunctionWithExecByCaller(t *testing.T) {
+
+	esdtData, _, _ := esdtDtaStorage.getESDTDigitalTokenDataFromSystemAccount(tokenKey, defaultQueryOptions())
+	require.Equal(t, tokenMetaData, esdtData.TokenMetaData)
+	require.Equal(t, esdtData.Value, quantity)
+
+	esdtDataBytes := vmOutput.Logs[0].Topics[3]
+	var esdtDataFromLog esdt.ESDigitalToken
+	err = nftCreate.marshaller.Unmarshal(&esdtDataFromLog, esdtDataBytes)
+	require.Nil(t, err)
+	require.Equal(t, esdtData.TokenMetaData, esdtDataFromLog.TokenMetaData)
+
+	// Simulate one extra call with only one uri
+	vmInput = &vmcommon.ContractCallInput{
+		VMInput: vmcommon.VMInput{
+			CallerAddr: sender.AddressBytes(),
+			CallValue:  big.NewInt(0),
+			Arguments: [][]byte{
+				[]byte(token),
+				quantity.Bytes(),
+				[]byte(name),
+				big.NewInt(int64(royalties)).Bytes(),
+				hash,
+				attributes,
+				uris[0],
+				nonce.Bytes(),
+				originalCreator,
+			},
+		},
+		RecipientAddr: sender.AddressBytes(),
+	}
+	vmOutput, err = nftCreate.ProcessBuiltinFunction(sender, nil, vmInput)
+	require.Nil(t, err)
+	require.NotNil(t, vmOutput)
+}
+
+func TestEsdtNFTCreate_ProcessBuiltinFunctionCrossChainTokenErrorCases(t *testing.T) {
+	t.Parallel()
+
+	accounts := createAccountsAdapterWithMap()
+	enableEpochsHandler := &mock.EnableEpochsHandlerStub{
+		IsFlagEnabledCalled: func(flag core.EnableEpochFlag) bool {
+			return flag == ValueLengthCheckFlag || flag == SaveToSystemAccountFlag || flag == CheckFrozenCollectionFlag
+		},
+	}
+	esdtDtaStorage := createNewESDTDataStorageHandlerWithArgs(&mock.GlobalSettingsHandlerStub{}, accounts, enableEpochsHandler)
+	ctc, _ := NewCrossChainTokenChecker(nil, getWhiteListedAddress())
+	esdtRoleHandler, _ := NewESDTRolesFunc(marshallerMock, ctc, false)
+
+	args := createESDTNFTCreateArgs()
+	args.CrossChainTokenCheckerHandler = ctc
+	args.RolesHandler = esdtRoleHandler
+	args.Accounts = esdtDtaStorage.accounts
+	args.EsdtStorageHandler = esdtDtaStorage
+
+	nftCreate, _ := NewESDTNFTCreateFunc(args)
+	address := []byte("whiteListedAddress")
+	userSender := []byte("userAccountAddress")
+	sender := mock.NewUserAccount(address)
+
+	t.Run("invalid num of args, penultimate arg is not nonce", func(t *testing.T) {
+		vmInput := &vmcommon.ContractCallInput{
+			VMInput: vmcommon.VMInput{
+				CallerAddr: sender.AddressBytes(),
+				CallValue:  big.NewInt(0),
+				Arguments: [][]byte{
+					[]byte("sov1-TOKEN-abcdef"),
+					big.NewInt(2).Bytes(),
+					[]byte("name"),
+					big.NewInt(int64(100)).Bytes(),
+					[]byte("12345678901234567890123456789012"),
+					[]byte("attributes"),
+					[]byte("uri1"),
+				},
+			},
+			RecipientAddr: sender.AddressBytes(),
+		}
+
+		vmOutput, err := nftCreate.ProcessBuiltinFunction(sender, nil, vmInput)
+		require.ErrorIs(t, err, ErrInvalidNumberOfArguments)
+		require.True(t, strings.Contains(err.Error(), "for cross chain"))
+		require.Nil(t, vmOutput)
+	})
+
+	t.Run("invalid num of args in exec on dest, penultimate arg is not nonce", func(t *testing.T) {
```

Comment:
here we are missing `antepenultimate nonce` and `penultimate creator`

Reply:
Refactored tests